c10/util/typeid.h

```diff
@@ -308,6 +328,22 @@ inline C10_TYPENAME_CONSTEXPR TypeMetaData _makeTypeMetaDataInstance() {
           typeName};
 }
 
+template <class T>
+inline TypeMetaData _makeScalarTypeMetaDataInstance(ScalarType scalarType) {
```

Comment:
why is this not constexpr anymore? if it was - we could have guaranteed that no SIOF occurs I think

Reply:
I don't think it guaranteed anything - e.g. the macro is a no-op for old GCC etc.